NBitcoin/IpExtensions.cs

```diff
@@ -138,6 +138,37 @@ public static bool IsRFC4843(this IPAddress address)
 			return (bytes[15 - 15] == 0x20 && bytes[15 - 14] == 0x01 && bytes[15 - 13] == 0x00 && (bytes[15 - 12] & 0xF0) == 0x10);
 		}
 
+		public static byte[] GetGroup(this EndPoint endpoint)
+		{
+			if (endpoint is IPEndPoint ip)
+			{
+				return ip.Address.GetGroup();
+			}
+
+			if (endpoint is DnsEndPoint dns)
+			{
+				if (dns.IsTor() || dns.IsI2P())
+				{
+					var vchRet = new List<byte>();
+					if (dns.IsTor())
+					{
+						vchRet.Add((byte)1);

```

Comment:
It looks to me that this value is the same as https://github.com/lontivero/NBitcoin/blob/af9bc279e57c9592c18c61e2e9fb783f6eff20fd/NBitcoin/IpExtensions.cs#L199. 

Where does this value come from?

Reply:
It comes from my mind. I think it is just a magical number and for that reason it should be as good as any other. I can be wrong here.

Reply:
Would it make sense to have:

```csharp
var vchRet = new List<byte>();
vchRet.Add((byte)3); // the same as in https://github.com/lontivero/NBitcoin/blob/af9bc279e57c9592c18c61e2e9fb783f6eff20fd/NBitcoin/IpExtensions.cs#L219
if (dns.IsTor())
{
	vchRet.Add((byte)1);
}
else if (dns.IsI2P())
{
	vchRet.Add((byte)2);
}
```

to preserve the info about group type (i.e. `3` denotes Tor & CJDNS) as in https://github.com/lontivero/NBitcoin/blob/af9bc279e57c9592c18c61e2e9fb783f6eff20fd/NBitcoin/IpExtensions.cs#L219. But I won't pretend I understand this really?

Reply:
Done.